ENH: Add option to set reference volume when exporting segmentation to file

Sunderlandkyl opened the pull request:
This commit adds a vtkMRMLVolumeNode selector to the file export dialog, which allows a reference geometry to be specified when exporting a segmentation volume to file.
See discussion here: https://discourse.slicer.org/t/segmentation-alignment/13449/15

lassoan commented on Libs/MRML/Core/vtkMRMLSegmentationNode.cxx:
```diff
@@ -1186,4 +1186,34 @@ void vtkMRMLSegmentationNode::SetLabelmapConversionColorTableNodeID(const char*
+  this->Superclass::OnNodeReferenceAdded(reference);
+  if (std::string(reference->GetReferenceRole()) == this->GetReferenceImageGeometryReferenceRole())
+    {
+    this->InvokeEvent(vtkMRMLSegmentationNode::ReferenceImageGeometryChangedEvent, reference->GetReferencedNode());
```
Is there a specific reason not to use InvokeCustomModifiedEvent (which can be blocked by Start/EndModify)?

lassoan reviewed (commented):
This looks good, thanks a lot! Just one small question in a comment inline.

Sunderlandkyl commented:
Updated to use InvokeCustomModifiedEvent. I also removed some code duplication from the PR by adding vtkSlicerSegmentationsModuleLogic::GenerateMergedLabelmapInReferenceGeometry().

lassoan reviewed (approved):
Great, thank you!
